src/sentry/migrations/0233_recreate_subscriptions_in_snuba.py

```diff
@@ -0,0 +1,69 @@
+# Generated by Django 2.2.24 on 2021-09-24 19:23
+
+import logging
+
+from django.db import migrations
+
+from sentry.snuba.models import QueryDatasets
+from sentry.snuba.tasks import _create_in_snuba, _delete_from_snuba
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def migrate_subscriptions(apps, schema_editor):
+    QuerySubscription = apps.get_model("sentry", "QuerySubscription")
+
+    for subscription in RangeQuerySetWrapperWithProgressBar(
+        QuerySubscription.objects.select_related("snuba_query").all()
+    ):
+        if subscription.subscription_id is not None:
+            subscription_id = None
+            try:
+                subscription_id = _create_in_snuba(subscription)
+            except Exception as e:
+                logging.exception(f"failed to recreate {subscription.subscription_id}: {e}")
+                continue
+
+            try:
+                _delete_from_snuba(
+                    QueryDatasets(subscription.snuba_query.dataset),
+                    subscription.subscription_id,
+                )
+            except Exception as e:
+                logging.exception(f"failed to delete {subscription.subscription_id}: {e}")
```

Comment:
Is it possible that creating new subscription at line 21 succeeded above but deleting the old subscription failed. If yes, wouldn't you have 2 subscriptions in that case. Should you delete the subscription created at line 21 over here then?

Reply:
Yeah that's a good idea.